src/Lookup/DispatchingTermLookup.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\EntityId;
+use Wikimedia\Assert\Assert;
+use Wikimedia\Assert\ParameterAssertionException;
+
+/**
+ * Wrapper around repository-specific TermLookups
+ * picking up the right lookup object for the particular input.
+ *
+ * @since 3.7
+ *
+ * @license GPL-2.0+
+ */
+class DispatchingTermLookup implements TermLookup {
+
+	/**
+	 * @var TermLookup[]
+	 */
+	private $lookups;
+
+	/**
+	 * @param TermLookup[] $lookups associative array with repository names (strings) as keys
+	 *                              and TermLookup objects as values. Empty-string key
+	 *                              defines lookup for the local repository.
+	 *
+	 * @throws ParameterAssertionException
+	 */
+	public function __construct( array $lookups ) {
+		Assert::parameter(
+			!empty( $lookups ) && array_key_exists( '', $lookups ),
+			'$lookups',
+			'must must not be empty and must contain an empty-string key'
```

Comment:
Not really, but that would require changes to the error handling so that it can handle that special case in a meaningful way.
In general, I could see that you want to treat all lookups equally "remote"… but I don't think that matches up with what was done in DM, does it?